pkg/core/resources.go

```diff
@@ -0,0 +1,32 @@
+package core
+
+import "github.com/dominikbraun/graph"
+
+type (
+	Construct interface {
```

Comment:
where do you suggest it goes? I didnt want to make a file just for the type and i add to this with functionality later on

Reply:
Probably compilation_document if it's going to be too small for a file on its own.

Reply:
then were back to having a circular dependency. They need to stay in the core package since config uses them and the compiler pkg uses config. 

Im fine putting them elsewhere but it needs to be within this package